src/dqe_idx.erl

```diff
@@ -227,6 +228,28 @@ metrics(Collection) ->
 
 %%--------------------------------------------------------------------
 %% @doc
+%% Returns a list of metric path suffixes of `Depth' that are prefixed
+%% by the given probe `Prefix', which can also be empty.
+%%
+%% For example:
+%% metrics(<<"collection">>, <<>>, 1) -> [<<"base">>].
+%% metrics(<<"collection">>, <<"base">>, 1) -> [<<"cpu">>].
+%% metrics(<<"collection">>, <<>>, 2) -> [[<<"base">>.<<"cpu">>]].
+%%
+%% Actual results are binary-encoded in `dproto' encoding.
+%% @end
+%%--------------------------------------------------------------------
+-spec metrics(Collection::collection(),
+              Prefix::metric(),
+              Depth::pos_integer()) ->
+                    {ok, [Metric::metric()]} |
```

Comment:
The return type and the examples don't match. from the return type it'd be `[binary()]` which matches the first two examples, the third example however would be `[[binary()]]`.

the third example feels 'more' correct to me, mixing returns between lists of binaries and lists of lists of binaries will create headache when querying (effectively the code would need to have two branches to handle them instead of one.